### ISSUE TITLE
Make disabling loadingIndicator possible and instead show content with a proper status_code like 404 errors instead of 200 nuxt loadingIndicator page..

### ISSUE THREAD

**Issue by piotrorlowski** [nuxt/nuxt.js#8373]
### Is your feature request related to a problem? Please describe.

When trying to access a nonexistent page in nuxt, the first thing that happens and is visible especially on slow networks is the spinning loading indicator. I wonder how to disable this loading indicator https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-loading-indicator as in the prod environment this page sometimes is getting cached by the server as a 'proper' response (it's response is 200) and it should be 404 and therefore should not be cached by the server. If such a page is cached the whole app gets broken for specific users.

### Describe the solution you'd like

Can you provide me with a solution to disable loadingIndicator or to at least enable executing some code before it get's triggered so proper error status_code and conttent_type is attached to the server response? 

### Describe alternatives you've considered

I tried to do some error handling in middleware, we also have dedicated error pages that use nuxt error layout component however code isn't executed in middleware, nor error layout page gets loaded because loadingIndicator page is getting cached and then served by the server and whole application is getting broken.

**Comment by danielroe** [comment]
@piotrorlowski Have you tried disabling the loading indicator as described in the link you posted?

**Comment by piotrorlowski** [comment]
@danielpost I tried providing 'false' or an empty string, but both didn't work at least yesterday when I was investigating this issue. Still I was having the default Nuxt loading indicator page (200) before getting my javascript code executed.

However I tried today and it seems it worked, so maybe my instance didn't get updated or some other cache issue. Anyway thanks and sorry for the rising unnecessary alarm.

Anyway I am still getting 200 status_code and text/html response without loadingIndicator - is there a way in nuxt to get 404 status_code? Or is it more of server configuration thing in such case?
